### ISSUE TITLE
Slack pulse issue with 0.36.0: Slack channel named `metabase_files` is missing & other channels missing from Pulse dropdown with > 1000 Slack channels

### ISSUE THREAD

**Issue by lukeman** [metabase/metabase#12978]
**Describe the bug**

After an upgrade to v0.36.0 with an existing Slack bot integration, Metabase is unable to find the #metabase_files channel to store uploaded images and Slack pulses fail to send. The channel exists and is public, but it's not in the list of channels that Metabase seems to know about. Emails for the same pulses still go out fine.

There's an additional issue I noticed in testing to create this issue where I'm unable to find some channels in the channels dropdown when trying to create a pulse (the list that is present is not small, so it's for sure working to some degree). Our org has… a lot of channels (over 6,000 listed when I launch the channel browser in the app), so unless it's some sort of filtering issue, this makes me wonder if the partial list is related to paging.

I tested a clone of this environment with rc2 and Slack was working fine, so it seems like either the Slack API or MB integration might've changed after rc2 (I'm just now noticing #12921, which stands out as a PR of interest).

**Logs**
```
[c4926f1a-8415-437a-a5c9-857633e98308] 2020-07-22T11:01:39-04:00 ERROR metabase.task.send-pulses Error sending Pulse 11
[c4926f1a-8415-437a-a5c9-857633e98308] 2020-07-22T11:01:50-04:00 ERROR metabase.integrations.slack Slack channel named `metabase_files` is missing! Please create the channel in order to complete the Slack integration. The channel is used for storing graphs that are included in Pulses and MetaBot answers.
[c4926f1a-8415-437a-a5c9-857633e98308] 2020-07-22T11:01:50-04:00 ERROR metabase.task.send-pulses Error sending Pulse 9
[c4926f1a-8415-437a-a5c9-857633e98308] 2020-07-22T11:02:15-04:00 ERROR metabase.integrations.slack Slack channel named `metabase_files` is missing! Please create the channel in order to complete the Slack integration. The channel is used for storing graphs that are included in Pulses and MetaBot answers.
```

**To Reproduce**
Steps to reproduce the missing metabase_files behavior:
1. Upgrade to v0.36.0 with a Slack integration in an org with lots of channels
2. Existing pulses fail to send via Slack with the included error in the logs

Steps to reproduce the channel list behavior:
1. Set up a Slack integration in an org with lots of channels
2. Create a new Pulse with Slack posts
3. Try to find a channel that exists but isn't in the list

**Expected behavior**
Existing Slack Pulses should send and public channels should be findable in the channels dropdown when creating a new Pulse.

**Information about your Metabase Installation:**

- Your browser and the version: Edge, Safari
- Your operating system: macOS
- Your databases: BigQuery, Postgres, Presto
- Metabase version: 0.36.0
- Metabase hosting environment: Docker in Nomad
- Metabase internal database: Postgres

**Severity**
Seems critical for affected folks who rely on Slack pulses (and possibly Slackbot, although I'm not using that).

**Additional context**
I haven't read through the new implementation so this is purely a guess, but I'm reminded of some paging issues I've been bitten by with other Slack integrations I've written/used since Slack's limits were high enough at the time that you'd only really notice an issue with orgs that stretched past the thousands of entities (be it users or channels).

To that regard, I'm also guessing that the size of our org is at fault here. Would be good to hear if any small Slack orgs have also experienced issues.

**Comment by lukeman** [comment]
Some additional info:

* I set up a local clone and the behavior is the same, so I don't think it was an ephemeral issue with the API
* The channel list in the pulse creation form seems the same on both instances
* The (partial) channel list appears to be in order of channel creation, followed by a (partial) list of users in order of creation
* The list of users includes deactivated accounts

**Comment by camsaul** [comment]
I put a limit in there of 1000 items as a failsafe to make sure we didn't get into an infinite loop of fetching page after page of channels from the Slack API. Previously we only fetched the first page (so only the first 100) results. 

It looks like the limit is too small in reality so we can bump that limit. Unfortunately it doesn't look like the Slack API has a way to search for channels by name so it's not great that we'd have to do more than 10 API requests (the current limit) to find the `metabase_files` channel

**Comment by lukeman** [comment]
Makes sense. I mixed up the max in the integration thinking it was meant to be the limit that you can specify to Slack.

If Metabase were to pass a `limit` param of 1000 in the Slack API calls (that's the max last I checked), that would dramatically reduce the number of API calls needed for larger orgs. That limit has worked fine in some of my scripts against the API in the past.

**Comment by camsaul** [comment]
So after looking at the PR a bit I don't think we meant to fetch archived things or users for that matter

https://github.com/metabase/metabase/pull/12921/files#diff-3fb52ec8afa1ba87c164bf7af2e371b8R99-R104

I thought this was working as intended when I tested it but for whoever tackles this, it might be a good place to start (verifying this does what's intended)

**Comment by lukeman** [comment]
> Previously we only fetched the first page (so only the first 100) results.

I considered that maybe I had just gotten lucky in the past, but after rolling back to 0.35.4 I just confirmed that the prior implementation pulls in over 19,000 values in the Slack field options from the `form_input` API call and not just the first 100.

**Comment by flamber** [comment]
@lukeman Then it sounds like #11735. Changing priority, since it's not a regression.

**Comment by lukeman** [comment]
@flamber Sorry if my message wasn't clear, but I actually meant to confirm that it is a regression (the prior version pulls in the entire list of 19,000+, whereas 0.36.0 only pulls in 2000). The issue discussed in #11735 occurs because prior versions do process those large lists but can occasionally hit a read timeout issue (I wasn't aware of that open issue, but I've seen that bug off and on in the past).

**Comment by camsaul** [comment]
Fixed by #13105
